src/node_http2_state.h

```diff
@@ -0,0 +1,116 @@
+#ifndef SRC_NODE_HTTP2_STATE_H_
```

Comment:
I had to move most of the stuff here out of `node_http2.cc` because `http2_state` was forward declaration, and I was seeing a compiler warning about calling delete on a forward declaration.  I initially tried to introduce a `#include "node_http2.h"` into `env.h`, but that was causing cycles.